Provide more information about the 'Support' filter to avoid confusion

Maran23 opened the pull request:
Fixes https://github.com/godotengine/godot/issues/70598
-> May help to avoid confusion as written down in the issue.

| Before | After |
|-|-|
| ![image](https://user-images.githubusercontent.com/66004280/210149493-2835afa6-ea10-4a7d-8c72-75205aee0ee4.png) | ![image](https://user-images.githubusercontent.com/66004280/210151700-855375f0-4502-4c7d-bc89-ee161b41c67e.png) |
| ![image](https://user-images.githubusercontent.com/66004280/210151493-c5977395-8937-4080-a04e-cb0fba6b7fcb.png) | ![image](https://user-images.githubusercontent.com/66004280/210151476-c716b5b8-71c9-4f71-be94-d04a3eb7636c.png) |

Maran23 committed:
Provide more information about the 'Support' filter when no result was found

Avoid possible confusion by showing more information about the 'Support' filter

Calinou commented:
I would suggest explicitly listing the Support levels that were chosen. For example:

> No results compatible with Godot 4.0 for support levels "Official".
> Check the enabled support levels using the Support button in the top-right corner.

Maran23 commented:
> This is a good start, but I suggest explicitly listing the support levels that were chosen.

Done. I also added this to the normal filter search result, as it can also happen there.
I also updated the screenshots.
Open for feedback. :)

pwab commented:
Hey @Maran23 thanks for this little but helpful addition. I also was confused about this message and apart from the missing explanation this could be triggered two other things:

1. I did not realize that the support-filter button is in fact ... a button. It is styled like the labels (Sort, Category, Site) without any border or highlight (color, shading) to make it look like a button. Also, all dropdowns in this window have this little triangle on the right side to indicate that the field is a dropdown element. The support-filter button does not have a triangle.
2. The naming 'Support' might also be confusing. When looking at it in this menu, my thought as a first time user is 'This must lead to the asset-lib support page'. But I have no clue what kind of renaming could help here.

Would it be possible to incorporate some minor UI changes next to the textual explanation to improve this dialog even further?

Maran23 commented:
> 1. I did not realize that the support-filter button is in fact ... a button. It is styled like the labels (Sort, Category, Site) without any border or highlight (color, shading) to make it look like a button. Also, all dropdowns in this window have this little triangle on the right side to indicate that the field is a dropdown element. The support-filter button does not have a triangle.

I agree. Unfortunately this is probably a bigger task, since this component (`MenuButton`) is used more often around the engine. So not only this location suffers from this.
![image](https://user-images.githubusercontent.com/66004280/210183281-8dad95e4-3cab-414c-abb4-fbfebb6d2f91.png)

We may want to open a proposal to improve the look and feel of `MenuButton`s.

> 2. The naming 'Support' might also be confusing. When looking at it in this menu, my thought as a first time user is 'This must lead to the asset-lib support page'. But I have no clue what kind of renaming could help here.

Not sure here too. I also find this name quite confusing. This is more likely some kind of category.

This PR is just a low hanging fruit, trying to improve the current state without much refactoring or more.

akien-mga commented on editor/plugins/asset_library_editor_plugin.cpp:
```diff
@@ -1242,15 +1248,28 @@ void EditorAssetLibrary::_http_request_completed(int p_status, int p_code, const
 				if (!filter->get_text().is_empty()) {
 					library_info->set_text(
-							vformat(TTR("No results for \"%s\"."), filter->get_text()));
+							vformat(TTR("No results for \"%s\" for support levels: %s."), filter->get_text(), support_list));
```
```suggestion
							vformat(TTR("No results for \"%s\" for support level(s): %s."), filter->get_text(), support_list));
```

akien-mga commented on editor/plugins/asset_library_editor_plugin.cpp:
```diff
@@ -1242,15 +1248,28 @@ void EditorAssetLibrary::_http_request_completed(int p_status, int p_code, const
 					// and no assets compatible with the new version have been published yet.
 					library_info->set_text(
-							vformat(TTR("No results compatible with %s %s."), String(VERSION_SHORT_NAME).capitalize(), String(VERSION_BRANCH)));
+							vformat(TTR("No results compatible with %s %s for support levels: %s.\nCheck the enabled support levels using the 'Support' button in the top-right corner."), String(VERSION_SHORT_NAME).capitalize(), String(VERSION_BRANCH), support_list));
```
```suggestion
							vformat(TTR("No results compatible with %s %s for support level(s): %s.\nCheck the enabled support levels using the 'Support' button in the top-right corner."), String(VERSION_SHORT_NAME).capitalize(), String(VERSION_BRANCH), support_list));
```

akien-mga commented on editor/plugins/asset_library_editor_plugin.cpp:
```diff
@@ -1242,15 +1248,28 @@ void EditorAssetLibrary::_http_request_completed(int p_status, int p_code, const
+						if (!support_list.is_empty()) {
+							support_list += ", ";
+						}
+						support_list += TTR(support_ttr_key[i]);
```
This won't work to extract the translation strings, `TTR()` needs to be used with a literal.
You should use `TTRGET()` here, and `TTRC()` in the Array.

Maran23 replied:
Ahh, I see, thanks! 
Done. (All your comments)

akien-mga commented on editor/plugins/asset_library_editor_plugin.cpp:
```diff
@@ -1510,9 +1529,9 @@ EditorAssetLibrary::EditorAssetLibrary(bool p_templates_only) {
-	support->get_popup()->add_check_item(TTR("Official"), SUPPORT_OFFICIAL);
-	support->get_popup()->add_check_item(TTR("Community"), SUPPORT_COMMUNITY);
-	support->get_popup()->add_check_item(TTR("Testing"), SUPPORT_TESTING);
+	support->get_popup()->add_check_item(TTR(support_ttr_key[SUPPORT_OFFICIAL]), SUPPORT_OFFICIAL);
+	support->get_popup()->add_check_item(TTR(support_ttr_key[SUPPORT_COMMUNITY]), SUPPORT_COMMUNITY);
+	support->get_popup()->add_check_item(TTR(support_ttr_key[SUPPORT_TESTING]), SUPPORT_TESTING);
```
Same comment as above, this won't work, should use `TTRGET()`.

akien-mga commented on editor/plugins/asset_library_editor_plugin.h:
```diff
@@ -234,6 +234,7 @@ class EditorAssetLibrary : public PanelContainer {
 	static const char *sort_key[SORT_MAX];
 	static const char *sort_text[SORT_MAX];
 	static const char *support_key[SUPPORT_MAX];
+	static const char *support_ttr_key[SUPPORT_MAX];
```
Could be `support_text` to be consistent with `sort_key`/`sort_text`.

akien-mga commented:
Thanks!
